uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/federatedlogin/utils/IntegrationUtilsOrchestrator.java

```diff
@@ -0,0 +1,352 @@
+                                               String url,
+                                               String username,
+                                               String firstName,
+                                               String lastName,
+                                               String email,
+                                               boolean verified,
+                                               String phoneNumber) {
+        ScimUser user = new ScimUser();
+        user.setUserName(username);
+        user.setName(new ScimUser.Name(firstName, lastName));
+        user.addEmail(email);
+        user.setVerified(verified);
+        user.setActive(true);
+        user.setPassword("secr3T");
+        user.setPhoneNumbers(Collections.singletonList(new PhoneNumber(phoneNumber)));
+        return client.postForEntity(url + "/Users", user, ScimUser.class).getBody();
+    }
+
+    public static ScimGroup createGroup(String token,
+                                        String zoneId,
+                                        String url,
+                                        ScimGroup group) {
+        RestTemplate template = new RestTemplate();
+        template.setErrorHandler(fiveHundredErrorHandler);
+        MultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+        headers.add("Accept", APPLICATION_JSON_VALUE);
+        headers.add("Authorization", "bearer " + token);
+        headers.add("Content-Type", APPLICATION_JSON_VALUE);
+        if (hasText(zoneId)) {
+            headers.add(IdentityZoneSwitchingFilter.HEADER, zoneId);
+        }
+        ResponseEntity<ScimGroup> createGroup = template.exchange(
+                url + "/Groups",
+                HttpMethod.POST,
+                new HttpEntity<>(JsonUtils.writeValueAsBytes(group), headers),
+                ScimGroup.class
+        );
+        return createGroup.getBody();
+    }
+
+
+    private static IdentityZone createOrchZone(RestTemplate client,
+                                               String url,
+                                               String id,
+                                               String subdomain,
+                                               IdentityZoneConfiguration config,
+                                               boolean active) throws Throwable {
+        OrchestratorZoneRequest orchestratorZoneRequest = new OrchestratorZoneRequest();
+        orchestratorZoneRequest.setName(id);
+        orchestratorZoneRequest.setParameters(new OrchestratorZone("adminsecret", subdomain));
+        //Create orch zone
+        ResponseEntity<OrchestratorZoneResponse> zone = client.postForEntity(url + "/orchestrator/zones", orchestratorZoneRequest, OrchestratorZoneResponse.class);
+        //Get orch zone
+        ResponseEntity<OrchestratorZoneResponse> zone1 = client.getForEntity(url + "/orchestrator/zones?name=" + id, OrchestratorZoneResponse.class, id);
```

Comment:
The call to get zone is redundant. Create zone already returns ResponseEntity<OrchestratorZoneResponse>; parse that instead.